### ISSUE TITLE
Make generated typescript usable for production code

### ISSUE THREAD

**Issue by mrbonezy** [ton-community/tact#13]
## Context
Dapps using a smart contract developed in Tact can benefit from code generated by the compiler.
Currently, the generated typescript class relies on `ContractExecutor` that is compatible with tests only.

## Proposed change
Most helpful would be to expose all message builders and getters in a way that doesn't rely on ContractExecutor.
That way both getters and messages can be reused for production code.

Other:
1. Split `send` to a function per message type. Would that enable eliminating `$$type`?
2. Each type of message gets a `[msg]Body` function (includes both comment-based messages, and for other messages this just means renaming from `pack[Message]`
3. Remove `export` where possible, so only contract-specific functions are exported

**Comment by ex3ndr** [comment]
No, all this types need to be serialized separately since they could be reused by different types of code anywhere outside of calls, like storing them in db, this doesn't always mean storing messages, but also some structs separately.

**Comment by ex3ndr** [comment]
Also ContractExecutor is a good abstraction and can work with a cloud and with a local one. You can easily write your own implementation.

**Comment by mrbonezy** [comment]
> Also ContractExecutor is a good abstraction and can work with a cloud and with a local one. You can easily write your own implementation.

Can we make it an interface and make the current one a concrete implementation?

**Comment by mrbonezy** [comment]
> No, all this types need to be serialized separately since they could be reused by different types of code anywhere outside of calls, like storing them in db, this doesn't always mean storing messages, but also some structs separately.

Is this referring to `$$type`? If so, there's no problem it exists actually, but a more convenient API would relieve the user of having to be aware of it. So perhaps the generator could add helper functions for constructing these types.

**Comment by ex3ndr** [comment]
I have worked last week to rewrite ton libraries and created a standard API for both API libraries and emulation one.

I have introduced the method `open` everywhere that allows you to open an instance of a `Contract` and allows you to easily work with contract using a simplest API possible. 

And bindings now are independent from executor.

Example: https://github.com/ton-community/tact-template/blob/5e645bbba5b077ddcc5fc6a1f785337f574b2b9c/sources/contract.spec.ts#L11
